test/parallel_api/algorithm/alg.nonmodifying/equal.pass.cpp

```diff
@@ -98,49 +98,84 @@ struct UserType
 };
 
 template <typename T>
-struct test_one_policy
+struct test_with_4_iters
 {
     template <typename ExecutionPolicy, typename Iterator1, typename Iterator2>
     void
-    operator()(ExecutionPolicy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, bool is_true_equal)
+    operator()(ExecutionPolicy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, Iterator2 last2, bool is_true_equal)
     {
-        using namespace std;
-
-        auto expected = equal(first1, last1, first2);
-        auto actual = equal(exec, first1, last1, first2);
-        EXPECT_EQ(expected, actual, "result for equal for random-access iterator, checking against ::std::equal()");
-
-        // testing bool
-        EXPECT_TRUE(is_true_equal == actual, "result for equal for random-access iterator, bool");
+        auto actual = ::std::equal(exec, first1, last1, first2, last2);
+        EXPECT_TRUE(is_true_equal == actual, "result for equal (4 iterators, without predicate) for random-access iterator, bool");
+#if CPP14_ENABLED
+        auto expected = ::std::equal(first1, last1, first2, last2);
+        EXPECT_EQ(expected, actual, "result for equal (4 iterators, without predicate) for random-access iterator, checking against ::std::equal()");
```

Comment:
Why do we need this check?

Reply:
This may check if we provided proper `is_true_equal` flag. Imagine if we provided `false`, the algorithm returned `false` but it should have returned `true`. I treat this "double mistake" as a rare case, so I suggest removing the check.